src/graph.rs

```diff
@@ -15,12 +15,47 @@ use std::collections::HashMap;
 use std::fmt::Write;
 use std::marker::PhantomData;
 
+#[derive(Clone)]
+struct ItemsPerSecond(HashMap<String, TimeSeries<u32, u32>>);
+
+impl ItemsPerSecond {
+    fn new() -> ItemsPerSecond {
+        ItemsPerSecond(Default::default())
+    }
+
+    fn initialize_or_increment(&mut self, key: &str, second: usize, value: u32) -> u32 {
+        if !self.0.contains_key(key) {
+            self.0.insert(key.to_string(), TimeSeries::new());
+        }
+        let data = self.0.get_mut(key).unwrap();
+        data.increase_value(second, value);
+        data.get(second)
+    }
+
+    #[allow(dead_code)]
+    fn insert(&mut self, key: &str, time_series: TimeSeries<u32, u32>) {
+        self.0.insert(key.to_string(), time_series);
+    }
+
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    fn get(&self, key: &str) -> Option<TimeSeries<u32, u32>> {
+        self.0.get(key).cloned()
+    }
+
+    fn get_map(&self) -> HashMap<String, TimeSeries<u32, u32>> {
+        self.0.clone()
+    }
```

Comment:
Not sure how performance critical this is, but might be good to inline these four functions?

(Might already be done, too)